php_phongo.c

```diff
@@ -1138,7 +1151,9 @@ mongoc_stream_t* phongo_stream_initiator(const mongoc_uri_t *uri, const mongoc_h
 	spprintf(&uniqid, 0, "%s:%d[%s]", host->host, host->port, mongoc_uri_get_string(uri));
 
 	MONGOC_DEBUG("Connecting to '%s'", uniqid);
+	zend_replace_error_handling(EH_SUPPRESS, NULL, &error_handling TSRMLS_CC);
```

Comment:
Since we craft a `bson_error_t` a few lines below if the stream creation fails, I was OK with suppressing errors here. My previous solution to throw meant that we'd get an extra exception on the stack, which then got collected as the previous exception when we inevitably thrown one later. The message and code ends up coming back to us from `php_stream_xport_create()`'s output parameters, so there's no loss of error reporting IMO.